test-files/test_proto.proto

```diff
@@ -139,4 +139,15 @@ message Wrapped {
 enum EnumAnnots {
   FOO = 0;
   BAR = 1 [deprecated=true];
+}
+
+message MapTest {
+  map<string, sint32> prim = 1;
+  map<int32, Wrapped> wrapped = 2;
+  // In theory you could do this because of how map is elaborated to a
+  // nested message under the hood. In practice it causes the reference impl to barf.
```

Comment:
Could you expand on this comment?  In particular, what issues do you see in the reference implementation?

Reply:
i thought i removed this...